doc/src/challenges/0010-string.md

```diff
@@ -1,4 +1,4 @@
-# Challenge X: Memory safety of String
+# Challenge 10: Memory safety of String
```

Comment:
Why 10?

Reply:
Other PRs are open that use up to 9.

Reply:
I wonder if we should just remove the number of the challenge.It's confusing since the section for this challenge is 3.7. Or we could change our approach to use a separate book for the challenges like we do for Kani RFC.

Reply:
It's nice to have numbers for the challenges so that we can easily refer to them. But I think they don't have to relate to the section number. Using separate books like the RFCs is also a good option if it's not too complicated.

Reply:
Per offline discussion, we decided to remove the numbered sections. Thanks Zyad!